packages/2d/src/scenes/Scene2D.ts

```diff
@@ -117,6 +117,22 @@ export class Scene2D extends GeneratorScene<View2D> implements Inspectable {
     }
   }
 
+  public transformMousePosition(
+    x: number,
+    y: number,
+    size?: {width: number; height: number},
+  ): Vector2 | null {
+    if (size !== undefined) {
+      x -= size.width / 2;
+      y -= size.height / 2;
+    }
+
+    x = Math.round(x);
+    y = Math.round(y);
+
+    return new Vector2(x, y);
```

Comment:
Coordinates shouldn't be transformed manually - you should use the view matrix instead.
This is the same logic we use for hit detection:
```suggestion
    return new Vector2(x, y)
      .scale(this.resolutionScale)
      .transformAsPoint(this.getView().localToParent().inverse());
```

Reply:
Applied.